Mauxite Construction Sprites

Flaborized opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To label this PR, add the label(s) without the prefixes surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->

 [FEAT]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

This PR makes it so some stuff made out of mauxite uses unique sprites instead of the material color being applied to the base. This includes:

- Walls + Rwalls
- Floors (Though floors with non-default patterns will currently use the patterned sprite. Might need to be fixed idk?)
- Lockers
- Stools
- Material sheets, rods, floortiles (Currently doesn't work, needs some code fix that I dunno how to do)
- Chairs
- Tables

Also change the default mauxite color to match these sprites


![](https://cdn.discordapp.com/attachments/1041726307584704532/1064805967025426492/image.png)



![](https://cdn.discordapp.com/attachments/1041726307584704532/1064802683313209394/image.png)


## Why's this needed? <!-- Describe why you think this should be added to the game. -->

I think this is a fun way to implement new building options, while also giving some aesthetic values to materials. Mauxite is just an arbitrary 'first' thing to work on, I plan to (and others can also) add stuff for other appropriate building materials, so you can have lots of options and uses for all the stuff mining can procure. Also the rusty industrial look could be useful for some prefabs which I've seen mentioned/been asked about on discord. 

Works with the new fancy $$ material tag stuff that Amy coded and pali helped me out with. Very strong and useful toolkit for making this kinda stuff with more or less no additional code.

## Changelog <!-- If necessary, put your changelog entry below. Otherwise, /please/ delete this entire section. -->
<!-- Put how you want to be credited in the changelog in place of CodeDude. -->
<!-- Use (*) for major changes and (+) for minor changes. See the contributor guide for details. For example: -->

```changelog
(u)Flaborized
(+)Added unique sprites for many construct-able things made out of mauxite sheets, for that industrial warehouse look. Includes walls, floors, tables, and chairs. Go build stuff with it! Mind the rust.
```

BatElite commented:
Interesting changelog

M-Earthfire commented:
How will this interact with alloys containing mauxite?

The sprites are lovely!

Flaborized commented:
> Interesting changelog :P

Whoops, fixed!

> How will this interact with alloys containing mauxite?
> 
> The sprites are lovely!

It won't change sprites of alloys, just pure mauxite. It *might* change them if you make an alloy that's named 'mauxite', but then it won't inherit the alloy color or anything. But why would one do that !

ZeWaka commented:
if it's wip,

WALPVRGIS commented:
Cool as hell sprites Flabo :000000 !!

Flaborized commented:
Added tables, chairs, changed the material color to match the rest of everything. (So objects without unique sprites will be closer looking to these bespoke ones) Also the sheets and rods and tiles weren't broken actually! With all that I think the PR is done, I dunno how to un-draft a PR (or if I can) but I think these are the features I wanna put in fully

![image](https://user-images.githubusercontent.com/53125172/213771009-21845ea8-a787-4ea9-aa1c-2abe7b2c069a.png)

Flaborized commented on code/modules/materials/Mat_Materials.dm:
```diff
@@ -363,7 +363,7 @@ ABSTRACT_TYPE(/datum/material/metal)
 	name = "mauxite"
 	desc = "Mauxite is a sturdy common metal."
-	color = "#574846"
+	color = "#534747
```
Whoops! Fixed, thanks for pointing that out :shelterfrog:
